scarlet/bbox.py

```diff
@@ -202,58 +169,22 @@ def insert_into(self, image, sub):
         return image
 
     @property
-    def C(self):
-        """Number of channels in the model
-        """
-        return self.shape[0]
-
-    @property
-    def Ny(self):
-        """Number of pixel in the y-direction
-        """
-        return self.shape[1]
-
-    @property
-    def Nx(self):
-        """Number of pixels in the x-direction
-        """
-        return self.shape[2]
-
-    @property
-    def front(self):
-        """Minimum z value
+    def D(self):
+        """Dimensionality of this BBox
         """
-        return self.origin[0]
+        return len(self.shape)
 
     @property
-    def bottom(self):
-        """Minimum y value
+    def start(self):
+        """Tuple of start coordinates
         """
-        return self.origin[1]
+        return self.origin
 
     @property
-    def left(self):
-        """Minimum x value
+    def stop(self):
+        """Tuple of stop coordinates (last is included)
         """
-        return self.origin[2]
-
-    @property
-    def back(self):
-        """Maximum y value
-        """
-        return self.origin[0] + self.shape[0]
-
-    @property
-    def top(self):
-        """Maximum y value
-        """
-        return self.origin[1] + self.shape[1]
-
-    @property
-    def right(self):
-        """Maximum x value
-        """
-        return self.origin[2] + self.shape[2]
+        return tuple(o + s for o, s in zip(self.origin, self.shape))
```

Comment:
It's probably worth making `Box` immutable and caching constants like this that will be used frequently.

Reply:
I don't think that box needs to be immutable when both `shape` and `origin` are tuples. I'll cache the `stop` values.

Reply:
If they aren't immutable then it's possible for `origin` and `shape` to be overwritten by a user, which would make a cached `stop` incorrect.  So we either need to make boxes immutable (which is what most codes do) or define setters for `origin` and `shape` that will update `stop` if either the `shape` or `origin` changes.

Reply:
That's why I coded it as property

Reply:
I prefer the latter option of keeping this flexibility.

Reply:
I think @herjy, you refer to the code as is, i.e. with a tuple that is generated on the fly. So do I. I don't think this is where we spend most of our time. If so, I'm happy to precompute it.